contents/haskell/Haskell.md

```diff
@@ -0,0 +1,195 @@
+its expressive power.
+
+While Haskell has its roots in academia, its emphasis on purity and
+side-effect-free computation makes it a valuable asset in software engineering
+contexts. Programs written in Haskell tend to be easy to test, refactor and
+debug, with the compiler usually catching all bugs before the program can even
+be compiled and run. Consequently, Haskell codebases are extraordinarily stable.
+
+Here's an example of a Haskell program that reads a string of numbers, prints
+the sum of the numbers and repeats the process until the string `"quit"` is
+entered. This shall serve as our Hello World.
+
+```haskell
+-- the entry point of the program
+main :: IO ()
+main = do
+    str <- getLine
+    if str == "quit"
+        then return ()
+        else do
+            let sumOfNumbers = sumAllNumbersInString str
+            putStrLn $ show sumOfNumbers
+            main
+
+sumAllNumbersInString :: String -> Int
+sumAllNumbersInString str = sumAll $ extractInts $ tokenize str
+
+-- sums up a list of integers using a higher-order function (the left fold)
+sumAll :: [Int] -> Int
+sumAll = foldl (+) 0
+
+-- convert each string of digits in a list to an actual integer
+extractInts :: [String] -> [Int]
+extractInts = fmap read
+
+-- split string by spaces using a built-in function
+tokenize :: String -> [String]
+tokenize = words
+```
+
+Haskell is widely used in a whole range of industries, including banks,
+financial companies, technology companies and engineering companies use Haskell
+in a variety of systems. A comprehensive list may be found
+[here](https://wiki.haskell.org/Haskell_in_industry).
+
+
+# Getting Started
+
+## Installation
+
+### System-Wide Installation
+
+For new users, Haskell may be quickly and easily installed by downloading the
+[Haskell Platform](https://www.haskell.org/platform/) for their respective
+operating systems. The Haskell Platform contains many common and important
+Haskell libraries, in addition to the Glasgow Haskell Compiler (GHC).
+
+At the time of writing, the Haskell Platform has binaries available for the
+following operating systems:
+
+* Linux
+  * Debian
+  * Ubuntu
+  * Mint
+  * Red Hat
+  * Fedora
+  * Gentoo
+  * a generic binary package
+* MacOS
+  * Homebrew
+  * ordinary installer
+* Windows
+
+### Stack Installation
+
+For Haskell projects of significant size, it may be necessary to control the
+exact versions of the compiler and libraries used. For such use cases, the
+system-wide installation method above may prove unwieldy and inadequate. In
+cases like these, it may be preferable to have an entire Haskell environment
+just for that project, together with a curated set of libraries.
+
+In such a scenario, [Stack](https://www.haskellstack.org) may come in handy.
+Stack is a package manager of sorts for Haskell, similar to NPM. Installation
+instructions may be found in the
+[Stack Documentation](https://docs.haskellstack.org/en/stable/README/), and is
+fairly standard.
```

Comment:
No. But I put it in there because the moment the learner wants to create an actual project, Stack comes in handy.